pyanaconda/kickstart.py

```diff
@@ -397,25 +418,30 @@ def execute(self, storage, ksdata, instClass):
                                            msg=(_("Requested boot drive \"%s\" doesn't exist or cannot be used.")
                                                 % self.bootDrive)))
         elif len(self.driveorder) >= 1:
-            log.debug("Bootloader: use '%s' first disk from driveorder as boot drive",
-                      self.driveorder[0])
+            log.debug("Bootloader: use '%s' first disk from driveorder as boot drive, dry run %s",
+                      self.driveorder[0], dry_run)
             self.bootDrive = self.driveorder[0]
         else:
-            boot_drive = None
             # Try to find /boot
-            for part in ksdata.partition.partitions:
-                if part.mountpoint == "/boot":
-                    device_match = device_matches(part.disk, devicetree=storage.devicetree)
-                    if len(device_match) == 1 and device_match[0] in disk_names:
-                        log.debug("Bootloader: use /boot partition '%s' as boot drive", device_match[0])
-                        boot_drive = device_match[0]
-                    break
-            else: # Nothing was found use first disk from Blivet
-                log.debug("Bootloader: fallback use first disk return from Blivet '%s' as boot drive",
-                          disk_names[0])
-                boot_drive = disk_names[0]
-
-            self.bootDrive = boot_drive
+            #
+            # This method is executed two times. Before and after partitioning.
+            # In the first run, the result is used for other partitioning but
+            # the second will be used.
+            try:
+                boot_dev = storage.mountpoints["/boot"]
+                boot_drive = boot_dev.parents[0].name
```

Comment:
The first parent of a device mounted at ``/boot`` doesn't necessarily have to be a disk. E.g. in case of RAID it would be a partition. You might need to go through the ``ancestors`` attribute here and grab the first disk from there.